ultralytics/engine/trainer.py

```diff
@@ -409,7 +409,7 @@ def _do_train(self, world_size=1):
                             break
 
                 # Log
-                mem = f"{torch.cuda.memory_reserved() / 1E9 if torch.cuda.is_available() else 0:.3g}G"  # (GB)
+                mem = f"{torch.cuda.memory_reserved() / 1E9 if torch.cuda.is_available() else torch.mps.driver_allocated_memory() / 1E9 if MACOS and torch.backends.mps.is_available() and self.device.type == 'mps' else 0:.3g}G"  # (GB)
```

Comment:
`torch.backends.mps.is_available()` is redundant as the method that selects the device already checks for that
https://github.com/ultralytics/ultralytics/blob/02e995383de4ee3754ceb96a993f91b070c71e77/ultralytics/utils/torch_utils.py#L210

Reply:
no "mps is not defined", or the likes,  issue that way with your non-mps-installed-but-macos setup?

Reply:
The selected device can only be `mps` when `torch.backends.mps.is_available()` is true already. Before beginning training ultralytics performs checks to select the correct device using the
https://github.com/ultralytics/ultralytics/blob/02e995383de4ee3754ceb96a993f91b070c71e77/ultralytics/utils/torch_utils.py#L124
method, which already checks for the mps backend before even allowing the device to be `mps`. It would otherwise fall back to the cpu by default.

Reply:
i tested on my mps setups and it works as you said, 'ill change it, just want to make sure there won't be for non mps-installs a "torch.backends has no 'mps'" kind of error?